solver/jobs.go

```diff
@@ -484,7 +486,7 @@ func (jl *Solver) Get(id string) (*Job, error) {
 	for {
 		select {
 		case <-ctx.Done():
-			return nil, errors.Errorf("no such job %s", id)
+			return nil, grpcerrors.WrapCode(errors.Errorf("no such job %s", id), codes.NotFound)
```

Comment:
Adding the grpc error into `control/gateway` makes sense to me, but adding it to `solver` is a bit more confusing. We don't do any other GRPC stuff in this package, so it's a bit weird to do it here.

As an alternative: we could create a new error, for `no such job`, then we can implement `Code() codes.Code` on that error, so we keep the exact code for this error in the same place for wherever we use it, and it makes sense to use it directly in this package, since it doesn't directly refer to GRPC.